src/content/actions/receive-profile.js

```diff
@@ -364,51 +401,64 @@ export function receiveProfileFromFile(profile: Profile): ThunkAction {
   };
 }
 
-export function errorReceivingProfileFromFile(error: any): Action {
+export function errorReceivingProfileFromFile(error: Error): Action {
   return {
     type: 'ERROR_RECEIVING_PROFILE_FROM_FILE',
     error,
   };
 }
 
+function _fileReader(input) {
+  const reader = new FileReader();
+  const promise = new Promise((resolve, reject) => {
+    reader.onload = () => resolve(reader.result);
+    reader.onerror = () => reject(reader.error);
+  });
+
+  return {
+    asText() {
+      reader.readAsText(input);
+      return promise;
+    },
+
+    asArrayBuffer() {
+      reader.readAsArrayBuffer(input);
+      return promise;
+    },
+  };
+}
```

Comment:
This is not entirely correct as a user of this util could call both asText and asArrayBuffer, but this wouldn't work because the promise would be resolved/rejected already. We'd need a different promise for each call but this makes things a little more complex and I'm not sure this is useful.